### DISCUSSION TITLE
Setup Multiple DUs with 1 CU

### DISCUSSION THREAD

**Discussion by hpate063** [srsran/srsRAN_Project#683] [General Help] [answered]
Hi, 

I am wondering if I can setup multiple DUs (all connected to one CU). I see mentions of using cell configurations to do this but I am wondering if there is any documentation which explains how to set this up clearly. If so, can someone please share? If not, can someone share any guidelines?
(Note: It is okay if UE can not handover between the 2 cells)

Further, can I connect these different DUs to the same RU (USRP) or do I need different RUs?

**Comment by pgawlowicz** [comment]
you can check our HO tutorial: https://docs.srsran.com/projects/project/en/latest/tutorials/source/handover/source/index.html

**Reply by hpate063** [reply]
Thanks. This is helpful.

**Reply by s21sm** [reply]
Hi @pgawlowicz ,
     Thanks for the HO tutorial. I followed the given instruction. My connection with COTS UE is ok, I can browse the Internet. I changed tx / rx gain, UE physical location, but hand over never triggered. The problem that I am facing is if I put `a3_report_type: rsrp `in the configuration file, gNB can not parse it, says `INI was not able to parse a3_report_type`. In [srsRAN code](https://github.com/srsran/srsRAN_Project/blob/main/configs/mobility.yml) I found they are using `event_triggered_report_type: a3`. I used that, and gNB parsed that. Here is my configuration file and gNB output.

[x310.txt](https://github.com/user-attachments/files/17291591/x310.txt)
[gNB.txt](https://github.com/user-attachments/files/17291611/gNB.txt)

Kindly help!

**Reply by pgawlowicz** [reply]
There was a change in the configuration structure. We are currently updating the docs. Please check here:  https://github.com/srsran/srsRAN_Project/blob/main/configs/mobility.yml

**Comment by frankist** [comment]
@hpate063, I got a bit confused by your question. I understood that you want to work with a CU/DU split instead of a monolithic gNB (CU and DU co-located) and you want to have 2 or more DUs connected to a single CU, correct? If that's the case, you cannot share one RU across multiple DUs. Each DU has its own RU. From the perspective of the connection between the DUs and the CU (F1 interface), the main thing you have to ensure is that the F1 bind IP address of the DUs does not collide if they are in the same machine, and that the DUs have different `gnb_du_id` and `pci` values. This is very easy to ensure this via the config file.

**Reply by hpate063** [reply]
Yes, correct. So two things here:
1) Ensure connection between DUs and CU: I can do that via config and that seems good. 
2) Connection between RU and DU: If an RU can not be shared across DUs, this is where I would like to inquire further. 

I want to have a system where 1 RU can be connected to multiple DUs. It is okay if it is 1 DU divided into 2 logical DUs or 2 separate DUs. The aim is to serve a UE via the RU-DU1-CU and move it's service to the RU-DU2-CU (while the session is active). 
I am thinking of implementing a module (possibly between the OFH and DU-low) to accomplish this. Another idea I am thinking of is to duplicate the creation of DU in software and manage states to realize the necessary functionality. 

I understand that my question is a bit abstract but I would like to understand the feasibility of either option and avoid one or the other in case the srsRAN architecture does not allow me. Can you tell me what option is feasible? Or if there an another way you suggest I can design?

**Reply by frankist** [reply, marked as answer]
If you do not have the requirement that the two DUs are in separate machines, what you are asking should be possible. We already support "RU-DU1-CU + RU-DU2-CU" when using the monolithic gNB app. You just need to increase the number of cells to 2 and the RU will treat the 2nd DU as an extra cell of a single DU. I don't see any conceptual impediment to do the same with the split CU/DU. The standalone DU app could also contain an "RU-DU1 + RU-DU2". 

I have to be honest though that I've not tested this yet, so I will have a look next week.

**Reply by hpate063** [reply]
That is great to hear. I will try to set it up as well and see what are the challenges. 

Thanks again and looking forward to your update.

**Reply by hpate063** [reply]
@frankist Were you able to test this setup? Can you provide me any guidelines on the same?

**Reply by Urojtariq** [reply]
Hi @hpate063 
I am also trying to make two DUs(connected to single CU), in the same machine. and I have tried all the optione mentioned above, but my  USRP(single) does not seems to connect with two DUs, I want to make sure that I am creating 2 DUs in the right format. Did you get any success in that, and Can you provide me with the guidelines.

**Reply by s21sm** [reply]
@Urojtariq quick question, what is your USRP model?

**Reply by Urojtariq** [reply]
B210

**Reply by s21sm** [reply]
> Handover requires the USRP being used to have a dual channel RF-frontend with independent RF chains. As a result, this tutorial uses an X310 as the RF-frontend.
> 
> Not all dual channel USRPs have independent RF chains, this means that, for example, a B200-series USRP is not suitable for this use case.
https://docs.srsran.com/projects/project/en/latest/tutorials/source/handover/source/index.html

**Reply by hpate063** [reply]
@Urojtariq 
I have a 200 series USRP and figured that I need 300 series to perform the necessary setup. 
I am yet to purchase and test that but I believe that should work as @s21sm mentioned.

**Reply by Urojtariq** [reply]
@s21sm @hpate063  Got it, Thanks a million.

**Comment by ulopezindra** [comment]
Hi,
I'm trying to use 1 CU along with 2 DUs and 2 RUs. Is this possible? Could you provide me with some guidance on how to do it?

```
                             DU - RU ----- UE    
                          /      
 CORE ----CU -
                          \ 
                             DU - RU ----- UE
```

**Reply by natthapolpetch** [reply]
This git https://github.com/srsran/srsRAN_Project/discussions/1256 try to do what you are looking for. Please see how to setup for a CU and both DUs here.
I am also doing this CU-DU split setup with 2 bladeRFs as RU connected to separate DU (DU1 and DU2), and one CU.
